x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/details_page_mappings_content.tsx

```diff
@@ -109,49 +129,54 @@ export const DetailsPageMappingsContent: FunctionComponent<{
   };
 
   const mappingsDefinition = extractMappingsDefinition(jsonData);
-  const { parsedDefaultValue } = useMemo<MappingsEditorParsedMetadata>(() => {
-    if (mappingsDefinition === null) {
-      return { multipleMappingsDeclared: true };
-    }
+  const { parsedDefaultValue, multipleMappingsDeclared } =
```

Comment:
Is it possible to extract this code into a separate file? also this is very similar to the logic in `x-pack/plugins/index_management/public/application/components/mappings_editor/mappings_editor.tsx`, so maybe both places could use a function for that?

Reply:
Updated.